### ISSUE TITLE
[86box added](http://86box.net/2022/11/26/86box-v3-11.html) Tandy PSSJ clone and TNDY ISA Tandy sound cards for non-Tandy machines. [The Tandy 1000 ISA sound card brainstorm thread](https://www.vogons.org/viewtopic.php?f=46&t=42521)

### ISSUE THREAD

**Issue by Torinde** [joncampbell123/dosbox-x#3950]
[86box added](http://86box.net/2022/11/26/86box-v3-11.html) Tandy PSSJ clone and TNDY ISA Tandy sound cards for non-Tandy machines. [The Tandy 1000 ISA sound card brainstorm thread](https://www.vogons.org/viewtopic.php?f=46&t=42521)

Is it possible to add such to DOSbox-X? That will resolve #3695?

PS. 
- Is the Discussion tab the place for feature requests now? I can't add a label though (in freshly generated issues it's possible)
- It seems the term "machine" in DOSbox(-X) actually means "video adapter"?

_Originally posted by @Torinde in https://github.com/joncampbell123/dosbox-x/discussions/3860_

**Comment by Torinde** [comment]
Adding these would be good but it might end up like GUS situation where you cannot swap from SB to GUS inside DosBox using `config -set` you have to start the environment with those values set. Regarding #3695, the OP wants DosBox to automagically set for the best environment. Having (yet another) option to switch if it requires a reboot/reload would not match their request.

There is also the issue mentioned that some titles look for certain Tandy markers which these emulated cards may not present, again requiring you still switch to the dedicated Tandy mode. Also some titles have slightly better graphics in Tandy mode (sort of EGA plus) but any game that could typically make use of VGA would likely by that point also understand SB, ADLIB and GUS which are all superior.

Personally, I'm all for more toys and switches to play with, if it gets added I would certainly try it out. 🙂

_Originally posted by @NebularNerd in https://github.com/joncampbell123/dosbox-x/discussions/3860#discussioncomment-4272551_
    
Definitely should be an option, not hard coded, yes - due to issues you mention.
The automagic part of 3695 is clear... what I meant is that by having such Tandy ISA cards it'll be possible to have VGA video + Tandy sound - for whatever situations that may be useful (I perceived part of the OP request in 3695 to be for that case).

_Originally posted by @Torinde in https://github.com/joncampbell123/dosbox-x/discussions/3860#discussioncomment-4273317_
    
Indeed, as discussed in the issue there were fringe cases of Tandy/VGA games and software, this would likely allow for those to be catered for (assuming no weird BIOS checks for Tandy-ness)

I'm constantly amazed by how clever emulation has become, it's a long way from PC-Task on the Amiga in the late 80's early 90's

_Originally posted by @NebularNerd in https://github.com/joncampbell123/dosbox-x/discussions/3860#discussioncomment-4274620_

**Comment by rderooy** [comment]
You can already enable Tandy sound on non-tandy machines by setting `tandy=on` in the config file. This is nothing new, and is the same with vanilla DOSBox.

```ini
[speaker]
tandy = on
```

Of course the value of this is pretty low as lots of Tandy games will only offer Tandy sound if the machine is actually a Tandy.

**Comment by Jkapp76** [comment]
Maybe we could imitate what the game is looking for to identify a tandy machine? sounds experimental, but could get a rare game or two to work.

**Comment by NebularNerd** [comment]
> You can already enable Tandy sound on non-tandy machines by setting `tandy=on` in the config file. This is nothing new, and is the same with vanilla DOSBox.
> 
> ```ini
> [speaker]
> tandy = on
> ```
> 
> Of course the value of this is pretty low as lots of Tandy compatible games will only offer Tandy sound if the machine is actually a Tandy.

Does that enable the Tandy DAC as well? From our conversation in #2380 the DAC will not function if the Soundblaster is still active, so `sbtype=none` is required as well which only works from the command line.

Part of the original issue is that the OP wants DosBox to figure out automatically the best way to play a game, as those of us who have had PC's since the good old days know there is almost no one setup that's best for all things.

**Comment by NebularNerd** [comment]
> Maybe we could imitate what the game is looking for to identify a tandy machine? sounds experimental, but could get a rare game or two to work.

It would need something to write specific markers to whatever ROM/BIOS/Memory location is needed, the possible issue there is if something else is using that space when booted as a regular PC. Assuming you did that there will no doubt be another game that looks for Tandy-ness just that little differently to still not detect.

**Comment by rderooy** [comment]
For a real retro PC, I can see some value in a Tandy soundcard, to allow **some** Tandy compatible games to have sound, as you cannot have every machine type under the sun in your cave.

But for emulation, I don't see any value in supporting an emulated Tandy soundcard on a non-Tandy machine type. Just set `machine=tandy` and you have a Tandy. Easy and straight-forward.

As to the Tandy DAC, yes you need to disable SB emulation as they have a resource conflict. I don't know if the Tandy DAC works with `machine=tandy`, I never use the option.

**Comment by RNMB15** [comment]
I find the idea TNDY ISA Tandy sound cards. in dosbox-x, good.

**Comment by Torinde** [comment]
> You can already enable Tandy sound on non-tandy machines by setting `tandy=on` in the config file. This is nothing new, and is the same with vanilla DOSBox.
> 
> ```ini
> [speaker]
> tandy = on
> ```
> 
> Of course the value of this is pretty low as lots of Tandy compatible games will only offer Tandy sound if the machine is actually a Tandy.

I think that answers the question - Tandy soundcard already can be emulated outside machine=Tandy ("machine", actually means videocard?). Whether games accept that depends on the game, not the emulator? Automatically finding optimal settings and such is a can of worms, IMHO...let's leave that for the other issue where it's discussed. 

What I see as open topic - the Tandy DAC - if it requires sbtype=none, which is possible ONLY from command line, then the "Tandy sound emulation" is incomplete in "standalone mode", e.g. can we have sbtype=none applied via .conf setting?

**Comment by rderooy** [comment]
The `machine=` setting is inherited from vanilla DOSBox. It is unfortunate that they chose to set videocards that way, instead of a separate option. But DOSBox forks cannot really change it without breaking backward compatibility with tons of DOSBox config files, shipped by retro DOS games on GOG, Steam, etc.

And of course you can set `sbtype=none` in the config file.

**Comment by NebularNerd** [comment]
> And of course you can set sbtype=none in the config file.

I really should mention this when I discuss these things, it's my habit to just run a base conf and switch these things from the command line so I always forget to highlight that anything I mention can be done in the conf as well.

Really all these sound cards would benefit from being set via a `soundcard=` line, you enter what you want and anything else is disabled. This would prevent resource conflicts between the various cards and make it easy to have `SB, GUS, Tandy, Adlib, Tandy Clone  etc...` setup and fully working, leaving `sbtype=` to just deal with the variations such as `SB16, VIBRA, AWE32 etc..`. However, this is much like the `machine=` issue that it would break compatibility with older conf's.

The only way I could see this sort of thing working is you would have to have a `[POWERUSER]` section, you have a trigger `POWERUSER=1`, this would then enable you to essentially 'build' a custom PC with fields like `BIOS=`, great for the old 5150 ROM you need to to make Casino work #3028 or setting a Tandy, PPC, whatever..., `GRAPHICS=` to pick your video adapter, `GRAPHICS2=` for dual video setups (would open another window), `SOUNDCARD=` for your soundcard and so on. Anything set here would blanket override any of the usual related .conf settings. Then you can try almost any combination your heart desires.

But that would require major work to be used by a very small minority I imagine.

**Comment by rderooy** [comment]
Yes the vanilla DOSBox config file, that the forks inherited, is a mess. From sound cards to machine types. But it is a legacy that is difficult to break with. Users will complain loudly (without bothering to checking docs or instructions) if their existing configs no longer work.

Regarding a single `soundcard=` option, that will not be ideal either as some people want to be able to setup games to use more than one soundcard. For instance, both a SB16 and a GUS. One for background music and the other for sound effects.

Ideally I would like machine type to be something like PC/XT, AT, PS/2, PCI, PC98 (or perhaps ISA8, ISA16, MCA, PCI would make more sense) and then CPU and GPU options.

But I don't see that happening, unless we can agree with the other forks to have a new format config file, and if that file is found, use it. And otherwise if an old style config file is found, have a parser that maps the old DOSBox syntax to the new format internally.

**Comment by NebularNerd** [comment]
I agree, totally breaking old conf's is no way to go there are 101 reasons why that would be a terrible idea.

However, the idea of a conf 2.0 or DosBox-x power user section be awesome, taking it to the extreme you could even go with slot definitions for emulating all sorts of things (a bit like WinUAE does) you could then have `isa1=`, `pci1=` , `mca1=` and so on, that would really allow you to build whatever exacting config you need, going further you may then want `isa1irq=` isa1dma=` and so on for non autoconfig cards and environment. All you do then is specify what you want in what slot, two soundcards? no problem, assign them to an empty slot and make sure they don't clash. 

But this is semi-crazy talk and would push DosBox more into the realm of becoming a proper Virtual PC style environment, I can imagine some of the issue posts with titles like `Why no work with 3 soundcards....` or `Pentium with 5150 Bios and VGA can't run Win95... `🤣

**Comment by rderooy** [comment]
I don't think DOSBox-X should compete with emulator like PCem or 86Box. If you want all kinds of crazy ISA/PCI card setups, use that.

The thing that attracts people to DOSBox, or one of its forks is the ease to get the vast majority of old PC games running. No need to create a HDD image, install DOS, load drivers for the mouse or CD-ROM and then try to tune it for max conventional memory.

**Comment by NebularNerd** [comment]
> I don't think DOSBox-X should compete with emulator like PCem or 86Box. If you want all kinds of crazy ISA/PCI card setups, use that.
> 
> The thing that attracts people to DOSBox, or one of its forks is the ease to get the vast majority of old PC games running. No need to create a HDD image, install DOS, load drivers for the mouse or CD-ROM and then try to tune it for max conventional memory.

As I said, semi-crazy talk 🤣 Personally I'm happy with DosBox and it's forks, it's lightweight, runs the vast majority of stuff I want and once you have it it tuned to your preferences then away you go. Some of the conf entries are a little tricky if you want to do something a little unconventional, but once you figure it out it's quite straightforward.

**Comment by Torinde** [comment]
@NebularNerd , DOSbox-X now can load ROM basic file via `ibm rom basic=` - I think that is the 5150 ROM for Casino? What is the "PPC" machine type you mention in your comment?

So, since Tandy sound can be selected and SB can be disabled, then basically the conclusion is "requested feature is already implemented"? Should I close this Issue?

I agree with both of you it would be nice to have .conf2/[POWERUSER] section and a backward-compatibility mechanism to/from .conf... although I think there are other things with higher priority - plus there is the difficulty in aligning the various forks, etc. - this topic has to moved to a new issue?

**Comment by NebularNerd** [comment]
> What is the "PPC" machine type you mention in your comment?

I meant `amstrad` I always think of my old PPC640 when I see that entry in the conf, they have a weird bios which some of their bundled software required. 😁 I was unaware of the `ibm rom` option being added, will give that a go and see if that cures the issue with Casino.

I think we kind of got a little off topic on this thread by discussing the thoughts regarding a .conf 2.0, your original post about adding the clone cards to the code base is still a good idea for people to have something new to play with. 😁

The only real issue with adding them is the documentation would need to highlight that while offering Tandy sound, Tandy games may not work due to expecting various pieces of Tandy-ness that a 'generic' PC would not have. Otherwise there will be threads opening up about 'enabling Tandy clone card but game x does not work...' which will invariably lead to a slew of replies saying '`machine=tandy` will solve this'

**Comment by Torinde** [comment]
OK, keeping the issue open - until somebody can check if the DAC works when machine=something-else-than-tandy.
